contracts/azorius/strategies/HatsProposalCreationWhitelistV1.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.19;
+
+import {IHatsProposalCreationWhitelistV1} from "../../interfaces/IHatsProposalCreationWhitelistV1.sol";
+import {IHats} from "../../interfaces/hats/IHats.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {ERC165} from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+
+abstract contract HatsProposalCreationWhitelistV1 is
+    IHatsProposalCreationWhitelistV1,
+    OwnableUpgradeable,
+    ERC165
+{
+    IHats public hatsContract;
+
+    /** Array to store whitelisted Hat IDs. */
+    uint256[] private whitelistedHatIds;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    /**
+     * Sets up the contract with its initial parameters.
+     *
+     * @param initializeParams encoded initialization parameters:
+     * `address _hatsContract`, `uint256[] _initialWhitelistedHats`
+     */
+    function setUp(bytes memory initializeParams) public virtual {
+        (address _hatsContract, uint256[] memory _initialWhitelistedHats) = abi
+            .decode(initializeParams, (address, uint256[]));
+
+        if (_hatsContract == address(0)) revert InvalidHatsContract();
+        hatsContract = IHats(_hatsContract);
```

Comment:
it can still be **invalid** hats contract even if it pass the check of zero address.

but I guess we has no reliable method to validate this? `!ERC165(_hatsContract).supportsInterface(type(IHats).interfaceId)` won't work because Hats contract itself didn't add that interfaceId to `supportsInterface`?

how about this?
```solidity
(bool success, ) = _hatsContract.staticcall(
     abi.encodeWithSignature("isWearerOfHat(address,uint256)")
);
require(success, "InvalidHatsContract");
```

Reply:
the zero address check is more about making sure that the parameter hasn't been accidentally omitted during the `setUp` call. considering that this `setUp` call is only made once per instance, and our own app is the one creating that calldata, I'm not overly concerned about this. i will probably not change this code.

Reply:
however, I will change the error name from `InvalidHatsContract` to `MissingHatsContract` which better reflects the intention of this zero address check.